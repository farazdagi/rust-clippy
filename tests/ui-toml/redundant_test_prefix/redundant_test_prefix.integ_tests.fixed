//@revisions: default integ_tests
//@[default] rustc-env:CLIPPY_CONF_DIR=tests/ui-toml/redundant_test_prefix/default
//@[integ_tests] rustc-env:CLIPPY_CONF_DIR=tests/ui-toml/redundant_test_prefix/in_integration_tests
//@compile-flags: --test
#![allow(dead_code)]
#![warn(clippy::redundant_test_prefix)]

fn main() {}

mod tests_no_annotations {
    use super::*;

    #[test]
    fn has_annotation() {
        //~[integ_tests]^ redundant_test_prefix
    }

    fn no_annotation() {}
}

#[test]
fn main_module_has_annotation() {
    //~[integ_tests]^ redundant_test_prefix
}

fn test_main_module_no_annotation() {}

#[cfg(test)]
mod tests {}
