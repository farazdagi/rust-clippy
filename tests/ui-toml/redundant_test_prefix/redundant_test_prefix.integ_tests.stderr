error: redundant `test_` prefix in test function name
  --> tests/ui-toml/redundant_test_prefix/redundant_test_prefix.rs:14:8
   |
LL |     fn test_has_annotation() {
   |        ^^^^^^^^^^^^^^^^^^^ help: consider removing the `test_` prefix: `has_annotation`
   |
   = note: `-D clippy::redundant-test-prefix` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::redundant_test_prefix)]`

error: redundant `test_` prefix in test function name
  --> tests/ui-toml/redundant_test_prefix/redundant_test_prefix.rs:22:4
   |
LL | fn test_main_module_has_annotation() {
   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider removing the `test_` prefix: `main_module_has_annotation`

error: aborting due to 2 previous errors

